### PR TITLE
FSC Espnet2 data preparation

### PR DESCRIPTION
**Pull request by siddhu001**

### CODE REVIEW

**Comment by sw005320** [comment]
Could you follow https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE#how-to-makeport-new-recipe?

**Comment by sw005320** [comment]
and https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md#132-espnet2-recipes

File: utils/text2token.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def get_parser():
         "-t",
         type=str,
         default="char",
-        choices=["char", "phn"],
+        choices=["char", "phn", "wrd"],
```
Comment (ftshijt) [inline utils/text2token.py:57]: ESPNet2 does not use functions in `utils` anymore. Meanwhile, I did not find where it is used in preparation. Could you remove this it not used?

File: egs2/fsc/asr1/run.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set="train"
+valid_set="valid"
+test_sets="test valid"
+
+asr_config=conf/tuning/train_asr_conformer7_n_fft512_hop_length256.yaml
```
Comment (ftshijt) [inline egs2/fsc/asr1/run.sh:12]: I would recommend you to use a soft link at `conf` as `train_asr.yaml` to make it standardized. (same for lm and decoding)

File: egs2/fsc/asr1/local/data.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    python3 local/data_prep.py ${fsc}
+    for x in test valid train; do
+        for f in text wav.scp utt2spk; do
+            sort data/${x}/${f} -o data/${x}/${f}
+        done
+        utils/utt2spk_to_spk2utt.pl data/${x}/utt2spk > "data/${x}/spk2utt"
+    done
+fi
+
+# if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
```
Comment (ftshijt) [inline egs2/fsc/asr1/local/data.sh:59]: Could you remove this if not used

File: egs2/fsc/asr1/db.sh [hunk 4]
```diff
@@ -0,0 +1 @@
+../../TEMPLATE/asr1/db.sh
```
Comment (ftshijt) [inline egs2/fsc/asr1/db.sh:1]: remember to update `db.sh`

**Comment by siddhu001** [comment]
Thanks a lot for sharing this Shinji. I will take a look and make the required edits.

**Review by ftshijt — COMMENTED** [review]
Many thanks! Here are some minor comments.

**Comment by ftshijt** [comment]
Also strongly recommend using `setup.sh` at TEMPLATE to standardized your effort

**Comment by siddhu001** [comment]
Currently with default ASR model configuration, we achieve a Test sentence error of 8.1 i.e. a total of 91.9% intent classification accuracy. 
Here is the graph of test set accuracy with increasing number of steps-
[test_accuracy_FSC.pdf](https://github.com/espnet/espnet/files/6820052/test_accuracy_FSC.pdf)

**Comment by sw005320** [comment]
@siddhu001, thanks for pasting the learning curve. It seems that overtraining does not happen. Please keep on working with that setup.

File: egs2/TEMPLATE/asr1/asr.sh [hunk 5]
```diff
@@ -301,6 +301,7 @@ else
     lm_token_type="${token_type}"
 fi
 
+echo ${asr_config}
```
Comment (ftshijt) [inline egs2/TEMPLATE/asr1/asr.sh:304]: Hi, we should remove debugging code here

**Comment by sw005320** [comment]
There are a couple of CI errors
https://github.com/espnet/espnet/pull/3352/checks?check_run_id=3138486281#step:7:78
https://github.com/espnet/espnet/pull/3352/checks?check_run_id=3138486275#step:7:293

**Comment by sw005320** [comment]
Note that you can run the CI test in your local by following https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md#4-unit-testing

File: egs2/fsc/asr1/db.sh (hunk 4 above)
Comment (siddhu001) [reply egs2/fsc/asr1/db.sh:1]: Done

**Comment by siddhu001** [comment]
Thanks a lot for all your feedback. I believe that all the comments on this pull request has been addressed now.

**Comment by sw005320** [comment]
LGTM!
Thanks, @siddhu001!
Please make a follow up PR to include
- [ ] put some results in egs2/fsc/asr1/README.md
- [ ] corpus description in egs2/README.md
- [ ] upload a model, and the corresponding config file
- [ ] add a description of the SLU task support in the main README.md

**Comment by siddhu001** [comment]
Sounds good! Thanks a ton for your help with this PR. I will open a followup PR to address these issues!

**Pull request merged**